src/services/MailpitApiContext.ts

```diff
@@ -12,118 +12,122 @@ export interface Email {
 }
 
 export class MailpitApiContext {
-    public context: APIRequestContext;
-  
-    constructor(context: APIRequestContext) {
-        this.context = context;
-    }
-  
-    /**
-     * Fetches email headers based on the recipient's email address.
-     * @param email - The email address of the recipient.
-     * @returns An Email object containing the email headers.
-     */
-    async getEmailHeaders(email: string): Promise<Email> {
-      const response = await this.context.get('/api/v1/search', {
-        params: {
-          kind: 'To.Address',
-          query: email,
-        },
-      });
-  
-      const responseJson = await response.json();
-    
-      return {
-        fromName: responseJson.messages[0].From.Name,
-        fromAddress: responseJson.messages[0].From.Address,
-        toName: responseJson.messages[0].To[0].Name,
-        toAddress: responseJson.messages[0].To[0].Address,
-        subject: responseJson.messages[0].Subject,
-        emailId: responseJson.messages[0].ID,
-      }
-    }
-  
-    /**
-     * Retrieves the body content of the latest email as an HTML string.
-     * @returns The HTML content of the latest email.
-     */
-    async getEmailBody(): Promise<string> {
-      const response = await this.context.get('view/latest.html');
-      const buffer = await response.body();
-      const htmlString = buffer.toString('utf-8');
-      return htmlString;
-    }
-  
-    /**
-     * Generates the full email content, combining headers and body.
-     * @param email - The email address to fetch headers for.
-     * @returns The full email content as a string.
-     */
-    async generateEmailContent(email: string): Promise<string> {
-      const headers = await this.getEmailHeaders(email);
-      const htmlTemplate = await this.getEmailBody();
-      
-      const headerSection = `
+  public context: APIRequestContext;
+
+  constructor(context: APIRequestContext) {
+    this.context = context;
+  }
+
+  /**
+   * Fetches email headers based on the recipient's email address.
+   * @param email - The email address of the recipient.
+   * @returns An Email object containing the email headers.
+   */
+  async getEmailHeaders(email: string): Promise<Email> {
+    const response = await this.context.get('/api/v1/search', {
+      params: {
+        kind: 'To.Address',
+        query: email,
+      },
+    });
+
+    const responseJson = await response.json();
+    const message = responseJson.messages[0];
+
+    return {
+      fromName: message.From.Name,
+      fromAddress: message.From.Address,
+      toName: message.To[0].Name,
+      toAddress: message.To[0].Address,
+      subject: message.Subject,
+      emailId: message.ID,
+    };
+  }
+
+  /**
+   * Retrieves the body content of the email as an HTML string.
+   * @param email - The email address of the recipient.
+   * @returns A promise that resolves to the HTML content of the email.
+   */
+  async getEmailBody(email: string): Promise<string> {
```

Comment:
Is that a breaking change?

Reply:
Actually, not, because we haven't used these methods before. 😄

Reply:
we not, but maybe external you don't know yet ;)

Reply:
Idk if someone uses this method. But there was a problem if we didn't add an email parameter. Because `getEmailBody()` brings the latest email content, we need to use it with `generateEmailContent(email: string)` to get the email body and its matching email headers. If we would like to use it alone to get the latest email body, like in the first version of the method, then we need to use `latest` as the email parameter. 
Docs: https://mailpit.axllent.org/docs/api-v1/view.html#get-/view/-ID-.html

Reply:
yes that is okay...we should then define within the commit message that this is a breaking change.

https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

Reply:
I added a commit message with description and breaking change footer